### DISCUSSION TITLE
Startup/Shutdown code execution?

### DISCUSSION THREAD

**Discussion by kyllerss** [sveltejs/kit#7156] [Ideas]
I would like to run some initialization code at server startup. The only advice I have found is to insert it into the ```hooks.server.ts``` handler. Is there no better way? Ideally we would have some startup handler that all startup initialization code would reside in. For that matter, it would also be helpful to have an idiomatic way of executing server shutdown code. 

Could someone point me towards any existing best practices that I may have missed? Is there a reason such a feature was never prioritized? Thank you!

**Comment by weepy** [comment]
You can write a startup script:

``` 
// /lib/startup.js
doInitialize()
```

then in any other server file you can put 

```
import "$lib/startup.js"

// rest
```

but I agree it doesn't feel very "solid". I think in an older version there was an "app.js".

**Reply by kyllerss** [reply]
I think it would go a long way to help organize code if there was one place one could check for all initialization code. Same would apply for any shutdown code.
